debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java

```diff
@@ -298,21 +301,64 @@
+            .withValidation(OracleConnectorConfig::validateLogMiningBufferType)
             .withImportance(Importance.LOW)
             .withDescription("The buffer type controls how the connector manages buffering transaction data." + System.lineSeparator() +
                     System.lineSeparator() +
                     "memory - Uses the JVM process' heap to buffer all transaction data." + System.lineSeparator() +
                     System.lineSeparator() +
-                    "infinispan - This option uses an embedded Infinispan cache to buffer transaction data and persist it to disk." +
-                    " Use the log.mining.buffer.location property to define the location for storing cache files.");
+                    "infinispan_embedded - This option uses an embedded Infinispan cache to buffer transaction data and persist it to disk." + System.lineSeparator() +
+                    System.lineSeparator() +
+                    "infinispan_remote - This option uses a remote Infinispan cluster to buffer transaction data and persist it to disk.");
 
+    @Deprecated
     public static final Field LOG_MINING_BUFFER_LOCATION = Field.create("log.mining.buffer.location")
             .withDisplayName("Location where Infinispan stores buffer caches")
             .withType(Type.STRING)
             .withWidth(Width.MEDIUM)
             .withImportance(Importance.LOW)
             .withValidation(OracleConnectorConfig::validateBufferLocation)
-            .withDescription("Path to location where Infinispan will store buffer caches");
+            .withDescription("(Deprecated) Path to location where Infinispan will store buffer caches");
+
+    public static final Field LOG_MINING_BUFFER_INFINISPAN_CACHE_TRANSACTIONS = Field.create("log.mining.buffer.infinispan.cache.transactions")
+            .withDisplayName("Infinispan 'transactions' cache configuration")
+            .withType(Type.STRING)
+            .withWidth(Width.LONG)
+            .withImportance(Importance.LOW)
+            .withValidation(OracleConnectorConfig::validateLogMiningInfinispanCacheConfiguration)
+            .withDescription("Specifies the XML configuration for the Infinispan 'transactions' cache");
+
+    public static final Field LOG_MINING_BUFFER_INFINISPAN_CACHE_COMMITTED_TRANSACTIONS = Field.create("log.mining.buffer.infinispan.cache.committed_transactions")
+            .withDisplayName("Infinispan 'committed-transactions' cache configuration")
+            .withType(Type.STRING)
+            .withWidth(Width.LONG)
+            .withImportance(Importance.LOW)
+            .withValidation(OracleConnectorConfig::validateLogMiningInfinispanCacheConfiguration)
+            .withDescription("Specifies the XML configuration for the Infinispan 'committed-transactions' cache");
+
+    public static final Field LOG_MINING_BUFFER_INFINISPAN_CACHE_ROLLBACK_TRANSACTIONS = Field.create("log.mining.buffer.infinispan.cache.rollback_transactions")
```

Comment:
That's a great question and one for which I don't have an answer for.  Looking at the code usages of the two it would appear we could consolidate these into a single cache, :+1:

Reply:
Merging these will mean upgrades from earlier versions using Infinispan will require a re-snapshot.